src/utils/geometry.js

```diff
@@ -0,0 +1,75 @@
+module.exports = {
+    // Add polygon composed by preClippedVertices to the `polygon.flat` array
+    clipPolygon(preClippedVertices, polygon, isHole) {
```

Comment:
maybe we could move this method to the `Polygon` class as `clip `and do

```js
clipped = polygon.clip(precliped, ishole);
```

maybe with some more refactor we can do only

```js
clipped = polygon.clip();
```